content/PreparedStatement/PreparedStatement.q.md

```diff
@@ -0,0 +1,1708 @@
+
+- It returns the boolean value `true`.
+
+- It holds the values `M` and `Male` in the SQL statement.
+
+- It calls the `execute()` method of the `$pstmt` object.
+
+- It assigns the specified values `M` and `Male` to the SQL statement.
+
+- It executes the argument `array('M', 'Male')` in the `execute()` method.
+
+
+:::
+
+
+:::
+
+/// type=REPL, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. <br />";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $pstmt = $conn->prepare("SELECT * FROM students WHERE gender = ?");
+    $pstmt->execute(array('Male'));
+    $data = $pstmt->fetchAll();
+    
+    foreach ($data as $row) {
+        echo $row['student_id'].$row['first_name'].$row['last_name'].$row['birth_date'].$row['gender']. ' ';
+    }
+?>
+```
+/// type=SS, answer=[4]
+
+In the statement `$pstmt->execute(array('Male'));` on line 5 of `student.php`, replace the value `Male` with `M`. Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database. ` and `1JohnSmith1999-02-10M 3BerryMeisner2002-09-09M 5DaronGuilliams2001-09-27M`.
+
+- It prints `Successfully connected to the database. ` and `2SamanthaDanes1999-10-12F 4VikkiFernando2005-05-17F 6AlisaElls1999-06-30F`.
+
+:::
+
+
++++
+
+
++++
+
+### Part 2: Knowledge Assessment
+
+/// type=MS, answer=[1,5]
+
+Which statements correctly describe `PDO::prepare()`?
+
+- It prepares an SQL statement.
+
+- It sends queries to databases.
+
+- It stores the values in an array.
+
+- It executes a prepared statement.
+
+- It returns a `PDOStatement` object.
+
+
+/// type=SS, answer=[2]
+
+What argument is passed in the `PDO::prepare()` method?
+
+- A DSN parameter.
+
+- An SQL statement.
+
+- An array of values.
+
+- An `echo` statement.
+
+- A `PDOStatement` object.
+
+
+/// type=SS, answer=[4]
+
+Which statement correctly describes `PDOStatement::execute()`?
+
+- It prepares an SQL statement.
+
+- It sends queries to databases.
+
+- It stores the values in an array.
+
+- It executes a prepared statement.
+
+- It returns a `PDOStatement` object.
+
+
+/// type=SS, answer=[3]
+
+What argument is passed in the `PDO::execute()` method?
+
+- A DSN parameter.
+
+- An SQL statement.
+
+- An array of values.
+
+- An `echo` statement.
+
+- A `PDOStatement` object.
+
+
+/// type=MS, answer=[3,4]
+
+What are the types of placeholders that can be used to prepare and execute an SQL statement?
```

Comment:
Please improve this statement. Maybe you can consider this:

> Which types of placeholders are supported by the prepared statement?

Or you may rephrase this base on your preference. Thanks.